src/utils.zig

```diff
@@ -4,13 +4,58 @@ const serialize = lib.serialize;
 const deserialize = lib.deserialize;
 const isFixedSizeObject = lib.isFixedSizeObject;
 const ArrayList = std.ArrayList;
+const Allocator = std.mem.Allocator;
+const sha256 = std.crypto.hash.sha2.Sha256;
+const hashes_of_zero = @import("./zeros.zig").hashes_of_zero;
+
+// SSZ specification constants
+const BYTES_PER_CHUNK = lib.BYTES_PER_CHUNK;
+const chunk = lib.chunk;
+const zero_chunk = lib.zero_chunk;
+
+/// Returns true if the type is a utils.List type
+pub fn isListType(comptime T: type) bool {
+    if (@typeInfo(T) != .@"struct") return false;
+
+    // Primary: check for explicit SSZ type marker
+    if (@hasDecl(T, "ssz_type_kind")) {
+        return T.ssz_type_kind == .list;
+    }
+
+    // Fallback: structural check
+    return @hasField(T, "inner") and
+        std.meta.hasFn(T, "sszEncode") and
+        std.meta.hasFn(T, "sszDecode") and
+        std.meta.hasFn(T, "append") and
+        std.meta.hasFn(T, "slice") and
+        !@hasField(T, "length");
```

Comment:
I don't think we should have that, because some other types might have exactly the same functions and be accepted when they shouldn't. So this is not more fool-proof than returning `false` when there was an issue in the recognition pattern and it should have been `true`.